plugins/modules/identity/keycloak/keycloak_clientscope.py

```diff
@@ -0,0 +1,475 @@
+      Keycloak API and its documentation at U(https://www.keycloak.org/docs-api/8.0/rest-api/index.html).
+
+    - Attributes are multi-valued in the Keycloak API. All attributes are lists of individual values and will
+      be returned that way by this module. You may pass single values for attributes when calling the module,
+      and this will be translated into a list suitable for the API.
+
+    - When updating a group, where possible provide the group ID to the module. This removes a lookup
+      to the API to translate the name into the group ID.
+
+
+options:
+    state:
+        description:
+            - State of the client_scope.
+            - On C(present), the group will be created if it does not yet exist, or updated with the parameters you provide.
+            - On C(absent), the group will be removed if it exists.
+        default: 'present'
+        type: str
+        choices:
+            - present
+            - absent
+
+    name:
+        type: str
+        description:
+            - Name of the client_scope.
+            - This parameter is required only when creating or updating the client_scope.
+
+    realm:
+        type: str
+        description:
+            - They Keycloak realm under which this group resides.
+        default: 'master'
+
+    id:
+        type: str
+        description:
+            - The unique identifier for this client_scope.
+            - This parameter is not required for updating or deleting a client_scope but
+              providing it will reduce the number of API calls required.
+
+    description:
+        type: str
+        description:
+            - Description for this client_scope.
+            - This parameter is not required for updating or deleting a client_scope.
+
+    protocol:
+        description:
+            - Type of client (either C(openid-connect), C(saml) or C(wsfed)).
+        choices: ['openid-connect', 'saml', 'wsfed']
+        type: str
+
+    protocol_mappers:
+        description:
+            - a list of dicts defining protocol mappers for this client.
+              This is 'protocolMappers' in the Keycloak REST API.
+        aliases:
+            - protocolMappers
+        type: list
+        elements: dict
+        suboptions:
+            protocol:
+                description:
+                    - This is either C(openid-connect), C(saml) or C(wsfed), this specifies for which protocol this protocol mapper
+                      is active.
+                choices: ['openid-connect', 'saml', 'wsfed']
+                type: str
+
+            protocolMapper:
+                description:
+                    - The Keycloak-internal name of the type of this protocol-mapper. While an exhaustive list is
+                      impossible to provide since this may be extended through SPIs by the user of Keycloak,
+                      by default Keycloak as of 3.4 ships with at least
```

Comment:
```suggestion
                      by default Keycloak as of 3.4 ships with at least:
```

Reply:
This change made pep8 through an error so I have removed it. 
Not sure if there is another way of fixing it.

Reply:
```suggestion
                      by default Keycloak as of 3.4 ships with at least:"
```